Kernel/semaphores.c

```diff
@@ -31,10 +33,23 @@ static int sem_free(Tsem sem) {
     return SEM_SUCCES;
 }
 
-static int isValidSemId(Tsem sem) {
+static int isValidSemId(TSem sem) {
     return sem>=0 && sem<MAX_SEMAPHORES && semaphores[sem]!=NULL;
```

Comment:
```suggestion
    return sem>0 && sem<MAX_SEMAPHORES && semaphores[sem]!=NULL;
```
(Or add 1 when you add a TSem to the resource namer)